File: 2022/day-04/sol.py
import re

with open('input1') as input1:
    ranges = input1.read().strip().split('\n')
    fully_containted, overlaped = 0, 0
    for range_pair in ranges:
        x1, y1, x2, y2 = map(int, re.split(r'-|,', range_pair))
        if (x1 >= x2 and y1 <= y2) or (x1 <= x2 and y1 >= y2):
           fully_containted += 1
        if (x1 >= x2 and x1 <= y2) or (y1 >= x2 and y1 <= y2) or (x2 >= x1 and x2 <= y1):
            overlaped += 1
    print(fully_containted, overlaped)

File: 2022/day-01/sol.go
package main

import (
	"fmt"
	"os"
	"bufio"
	"strings"
	"strconv"
)

func Max(nums ...int) int{
	res := nums[0]
	for _, num := range nums{
		if res < num {
			res = num
		}
	}
	return res
}

func main(){
	file, err := os.Open("./input1")
	if err != nil {
		panic(err)
	}
	defer file.Close()
	
	var ans1 int //, ans2 int32
	// var cals_per_elf []int
	reader := bufio.NewReader(file)
	buf := make([]byte, 3024)
	n, err := reader.Read(buf)
	if err != nil {
		panic(err)
	}
	fmt.Printf("bytes read: %v \n", n)
	
	fmt.Println(string(buf))
	for _, elfs := range strings.Split(string(buf), "\n\n") {
		cal_sum := 0
		for _, cal := range strings.Split(elfs, "\n") {
			if cal == "" {
				continue
			}
			cal_int, err := strconv.Atoi(cal)
			if err != nil {
				panic(err)
			}
			cal_sum += cal_int
		}
		// cals_per_elf = append(cals_per_elf, cal_sum)
		ans1 = Max(cal_sum, ans1)
	}

	fmt.Printf("%v \n", ans1)
}
